src/Components/CardboardList/CardboardList.types.ts

```diff
@@ -27,14 +27,18 @@ type IListItemBaseProps<T> = {
         customStyles?: IButtonStyles;
     };
     /** icon to render on the right side of the list item */
-    iconEnd?: {
-        name: IIconNames;
-        onClick?: (item: T) => void;
-    };
+    iconEnd?:
+        | {
+              name: IIconNames;
+              onClick?: (item: T) => void;
+          }
+        | ((item: T) => React.ReactElement);
```

Comment:
This is merely a question about the feasibility of something, feel free to resolve. Do you know if it is possible to type this with a specific component signature? i.e. `((item: T) => IconComponentType)`. If possible, does it make sense to have that constraint here?

Reply:
I'm not sure I follow the question. Right now this is passing back the list item itself. The return type is any React element you want. I believe `ReactElement` is actually a union type of JSXElement as well as some other stuff maybe? I know it's more broad than just JSXElement but I don't recall exactly why that's better, just that we used to use it on MSX portal. :)

Reply:
ohhh. idk if you can do that. But I'm also not sure we want to do that here since the whole idea is to be custom. Ideally you woulnd't try to render text here but I guess that's on you if you use it that way! 😜